test/read-stream.js

```diff
@@ -54,6 +54,88 @@ describe( 'BlockMap.ReadStream', function() {
 
   })
 
+  it( 'should not close the fd if autoClose = false', function( done ) {
+
+    var filename = path.join( __dirname, '/data/bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var fd = fs.openSync( filename, 'r' )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( null, blockMap, {
+      fd: fd,
+      autoClose: false,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
+        assert.ok( fs.fstatSync( readStream.fd ) )
+        fs.closeSync( readStream.fd )
+        done()
+      })
+
+  })
+
+  it( 'should start reading at `start`, if specified', function( done ) {
+
+    var filename = path.join( __dirname, '/data/padded-bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( filename, blockMap, {
+      start: 4096,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
```

Comment:
> and instead always (internally) calculate end = start + image_size ?

Since it will only read mapped blocks, it has a pre-calculated "end" at which it will stop regardless (which is not the same as the image size, but the last mapped block), unless the `end` option comes before that.
I don't see any benefit from calculating that, as it wouldn't have any use.